src/ProductConstructionService/ProductConstructionService.Api/DatabaseConfiguration.cs

```diff
@@ -25,5 +28,9 @@ public static void AddBuildAssetRegistry(this WebApplicationBuilder builder, str
                 sqlOptions.UseQuerySplittingBehavior(QuerySplittingBehavior.SingleQuery);
             });
         });
+
+        // inject the Kusto connection string in the Kusto Configuration section
+        builder.Configuration[_kustoConnectionStringKey] = builder.Configuration[PcsConfiguration.KustoConnectionString];
```

Comment:
I don't follow, why are we doing this?

Reply:
The `AddKustoClientProvider` method will read the Kusto section and register `KustoClientProviderOptions`. If we don't do this then the `KustoConnectionString` will be null, since it's not in the section

Reply:
Yeah, I meant why not just have the section right away?

Reply:
Because the service gets injected in the "base" Configuration section, under the same key as the secret name. So we need to inject it in the `Kusto` Section for it to get picked up. Or not use the `AddKustoClientProvider` method